include/utils/UserInfoUtil.php

```diff
@@ -3790,12 +3790,19 @@ function getPermittedModuleIdList() {
 /** Function to recalculate the Sharing Rules for all the users
   * This function will recalculate all the sharing rules for all the users in the Organization and will write them in flat files
  */
-function RecalculateSharingRules() {
+function RecalculateSharingRules($roleId) {
```

Comment:
We have to make this parameter optional:

```
function RecalculateSharingRules($roleId = 0) {
```
because it is currently called in a dozen places that would start emitting warnings if not